frontend/src/containers/ConfigStep/DateRangePicker/DateRangePickerGroup.tsx

```diff
@@ -6,20 +11,111 @@ import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { useAppDispatch, useAppSelector } from '@src/hooks/useAppDispatch';
 import { AddButton } from '@src/components/Common/AddButtonOneLine';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DateValidationError } from '@mui/x-date-pickers';
+import { useState, useEffect } from 'react';
+import sortBy from 'lodash/sortBy';
+import remove from 'lodash/remove';
+import get from 'lodash/get';
+import dayjs from 'dayjs';
 
-export const DateRangePickerGroup = () => {
+export enum SortType {
+  DESCENDING = 'DESCENDING',
+  ASCENDING = 'ASCENDING',
+  DEFAULT = 'DEFAULT',
+}
+
+export type ISortedDateRangeType = {
+  startDate: string | null;
+  endDate: string | null;
+  sortIndex: number;
+  startDateError: DateValidationError | null;
+  endDateError: DateValidationError | null;
+};
+
+const sortFn = {
+  DEFAULT: ({ sortIndex }: ISortedDateRangeType) => sortIndex,
+  DESCENDING: ({ startDate }: ISortedDateRangeType) => -dayjs(startDate).unix(),
+  ASCENDING: ({ startDate }: ISortedDateRangeType) => dayjs(startDate).unix(),
+};
+
+type IProps = {
+  sortType: SortType;
+  onChange?: (data: ISortedDateRangeType[]) => void;
+  onError?: (data: ISortedDateRangeType[]) => void;
+};
+
+const fillDateRangeGroup = <T,>(item: T, index: number) => ({
+  ...item,
+  startDateError: null,
+  endDateError: null,
+  sortIndex: index,
+});
+
+export const DateRangePickerGroup = ({ sortType, onError }: IProps) => {
   const dispatch = useAppDispatch();
   const dateRangeGroup = useAppSelector(selectDateRange);
   const isAddButtonDisabled = dateRangeGroup.length === MAX_TIME_RANGE_AMOUNT;
+  const [sortDateRangeGroup, setSortDateRangeGroup] = useState<ISortedDateRangeType[]>(
```

Comment:
sorted....

Reply:
thanks for reminder, change to `sortedDateRange`